### PR TITLE
enhance regex in fix_shebang method to fix more Python/Perl shebangs + avoid patching binary files

### PR DESCRIPTION
**Pull request by boegel**
fix for two problems that surfaced with @smoors' PR https://github.com/easybuilders/easybuild-easyconfigs/pull/8968

### CODE REVIEW

File: easybuild/framework/easyblock.py [hunk 1]
```diff
@@ -2158,6 +2158,7 @@ def package_step(self):
     def fix_shebang(self):
         """Fix shebang lines for specified files."""
         for lang in ['perl', 'python']:
+            shebang_regex = re.compile(r'^#![ ]*.*[/ ]%s[0-9.]*.*' % lang)
```
Comment (smoors) [inline easybuild/framework/easyblock.py]: should shebangs like `#!/usr/bin/env python3` and `#!/usr/bin/env python3.6` also be fixed?

Comment (boegel) [reply easybuild/framework/easyblock.py]: They are, see enhanced tests. The `[0-9.]*` part allows that the `python` is followed by `3` or `3.6`

Comment (smoors) [reply easybuild/framework/easyblock.py]: since there is a dot in the `[0-9.]*` I think this can be simplified: `r'^#![ ]*.*[/ ]%s.*' % lang`

Comment (boegel) [reply easybuild/framework/easyblock.py]: Yeah, good point, there's no point in using `[0-9.]*.*` indeed, might as well just use `.*`.

Comment (boegel) [reply easybuild/framework/easyblock.py]: fixed in a1cea138c

**Comment by smoors** [comment]
fix confirmed for https://github.com/easybuilders/easybuild-easyconfigs/pull/8968

**Pull request merged**